chains/solana/contracts/programs/cctp-token-pool/src/context.rs

```diff
@@ -0,0 +1,739 @@
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::get_associated_token_address_with_program_id;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+
+use base_token_pool::common::*;
+use ccip_common::seed;
+
+use crate::{CctpTokenPoolError, ChainConfig, MessageAndAttestation, State};
+
+const MAX_POOL_STATE_V: u8 = 1;
+const ANCHOR_DISCRIMINATOR: usize = 8;
+
+pub const TOKEN_MESSENGER_MINTER: Pubkey =
+    solana_program::pubkey!("CCTPiPYPc6AsJuwueEnWgSgucamXDZwBd53dQ11YiKX3");
+pub const MESSAGE_TRANSMITTER: Pubkey =
+    solana_program::pubkey!("CCTPmbSD7gX1bxKPAmg77w8oFzNFpaQiQUWD43TKaecd");
+
+pub const MESSAGE_SENT_EVENT_SEED: &[u8] = b"ccip_cctp_message_sent_event";
+
+#[derive(Accounts)]
+pub struct InitializeTokenPool<'info> {
+    #[account(
+        init,
+        seeds = [
+            POOL_STATE_SEED,
+            mint.key().as_ref(),
+        ],
+        bump,
+        payer = authority,
+        space = ANCHOR_DISCRIMINATOR + State::INIT_SPACE,
+    )]
+    pub state: Account<'info, State>,
+
+    pub mint: InterfaceAccount<'info, Mint>,
+
+    #[account(mut)]
+    pub authority: Signer<'info>,
+
+    pub system_program: Program<'info, System>,
+
+    // Token pool initialization only allowed by program upgrade authority. Initializing token pools managed
+    // by the CLL deployment of this program is limited to CLL. Users must deploy their own instance of this program.
+    #[account(constraint = program.programdata_address()? == Some(program_data.key()))]
+    pub program: Program<'info, crate::program::CctpTokenPool>,
+
+    #[account(constraint = program_data.upgrade_authority_address == Some(authority.key()))]
+    pub program_data: Account<'info, ProgramData>,
+}
+
+#[derive(Accounts)]
+pub struct Empty<'info> {
+    // This is unused, but Anchor requires that there is at least one account in the context
+    pub clock: Sysvar<'info, Clock>,
```

Comment:
Just to clarify the history of this restriction, since I was worried it might be interfering with the integration tests:

This is only the case when building with the `cpi` feature, correct? As per [this issue](https://github.com/solana-foundation/anchor/issues/1628). I believe anchor can handle an empty context otherwise.